.github/workflows/ci.yml

```diff
@@ -29,27 +29,61 @@ jobs:
           - 3.7
           - 3.8
           - 3.9
+        with-s3:
+          - 1
+          - 0
     steps:
       - name: Setup additional system libraries
         if: startsWith( matrix.os, 'ubuntu' )
         run: |
           sudo add-apt-repository multiverse
           sudo apt update
-          sudo apt install rar unrar
+          sudo apt install rar unrar libssl-dev libcurl4-openssl-dev zlib1g-dev
       - name: Setup Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Setup msbuild on Windows
+        if: matrix.with-s3 == 1 && matrix.os == 'windows-latest'
+        uses: microsoft/setup-msbuild@v1.1
+      - name: Set up Visual Studio shell
+        if: matrix.with-s3 == 1 && matrix.os == 'windows-latest'
+        uses: egor-tensin/vs-shell@v2
+        with:
+          arch: x64
       - name: Check out source repository
         uses: actions/checkout@v2
       - name: Install dependencies
         run: |
           pip3 install -r requirements.txt
           pip3 install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+          pip3 install cmake ninja pybind11 pybind11-global
+          echo "/home/runner/.local/bin" >> $GITHUB_PATH
       - name: Install test requirements
         run: pip3 install expecttest fsspec iopath==0.1.9 numpy pytest rarfile
+      - name: Install AWS-SDK-CPP on Windows for S3 IO datapipes
+        if: matrix.with-s3 == 1 && matrix.os == 'windows-latest'
+        run: |
+          git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp
+          cd aws-sdk-cpp
+          cmake -S . -B build -GNinja -DBUILD_ONLY="s3;transfer" -DENABLE_TESTING=OFF -DCMAKE_BUILD_TYPE=Release
+          cmake --build build --config Release
+          cmake --install build --config Release
+      - name: Install AWS-SDK-CPP on Non-Windows for S3 IO datapipes
+        if: matrix.with-s3 == 1 && matrix.os != 'windows-latest'
+        run: |
+          git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp
+          cd aws-sdk-cpp/
+          mkdir sdk-build
+          cd sdk-build
+          cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_ONLY="s3;transfer" -DAUTORUN_UNIT_TESTS=OFF
+          make
+          sudo make install
```

Comment:
Another follow up in my mind, does AWS provide docker images for such SDK then we don't have to download / compile it all the time?

Reply:
I don't think there's docker image for aws-sdk-cpp. It's too large and can be built and installed in very customized way on different systems.